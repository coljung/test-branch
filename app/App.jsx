import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';
import { ROUTE_HOME } from './constants/routes';
import Home from './Home';
import NotFound from './NotFound';
import Layout from './Layout';

const theme = createMuiTheme({
    palette: {
        primary: {
            main: '#ffffff',
        },
        secondary: {
            main: '#212121',
        },
        action: {
            active: 'rgba(255, 255, 255, 0.20)',
            hover: 'rgba(255, 255, 255, 0.15)',
            hoverOpacity: 0.15,
        },
    },
    typography: {
        fontSize: 12,
        useNextVariants: true,
    },
    shape: {
        borderRadius: 0,
    },
    overrides: {
        MuiDrawer: {
            paper: {
                backgroundColor: 'black',
            },
        },
        MuiButton: {
            textPrimary: {
                color: '#212121',
                '&:hover': {
                    textDecoration: 'underline',
                },
            },
        },
        MuiToolbar: {
            regular: {
                minHeight: '55px !important',
            },
        },
    },
    props: {
        MuiButtonBase: {
            disableRipple: true,
        },
    },
});

class App extends React.Component {
    createLayout = target => (
        <Layout appName='App Name'>
            {target}
        </Layout>
    );

    render() {
        return (
            <MuiThemeProvider theme={theme}>
                <CssBaseline />
                <BrowserRouter onUpdate={() => window.scrollTo(0, 0)}>
                    <Switch>
                        <Route path={ROUTE_HOME} exact render={() => this.createLayout(<Home />)} />

                        <Route render={() => this.createLayout(<NotFound />)} />
                    </Switch>
                </BrowserRouter>
            </MuiThemeProvider>
        );
    }
}

export default App;
