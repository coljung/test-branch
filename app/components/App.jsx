import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import { Layout, Icon } from 'antd';
import HeaderContent from './common/HeaderContent';
import NavigationMain from './common/NavigationMain';
import NotificationManager from '../notifications/NotificationManager';

const { Content, Header, Sider } = Layout;

export default class App extends Component {

    constructor(props) {
        super(props);
        this.state = { collapsed: true, showStoreModal: true };
    }

    toggle() {
        const collapsed = !this.state.collapsed;
        this.setState({ collapsed });
        clearTimeout(this.timer);

        // collapse after 8 seconds
        if (!collapsed) {
            this.timer = setTimeout(() => {
                this.setState({ collapsed: !collapsed });
            }, 8000);
        }
    }

    render() {
        return (
            <div className="store_layout">
                <Header>
                    <HeaderContent />
                </Header>
                <Layout>
                    <Content>
                        <main style={{ flex: 1, overflowY: 'auto', padding: 25 }}>
                            <ReactCSSTransitionGroup
                                    component="div"
                                    transitionName="example"
                                    transitionEnterTimeout={500}
                                    transitionLeaveTimeout={500}>
                                        {React.cloneElement(this.props.children, {
                                            key: this.props.location.pathname,
                                        })}
                            </ReactCSSTransitionGroup>
                            <NotificationManager />
                        </main>
                    </Content>
                </Layout>
            </div>
        );
    }

}

App.propTypes = {
    location: PropTypes.object,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.element),
        PropTypes.element,
    ]),
};
