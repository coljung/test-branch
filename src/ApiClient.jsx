import superagent from 'superagent';
import getApiUrl from './helpers';

const formatUrl = (path) => {
    const adjustedPath = path[0] !== '/' ? `/${path}` : path;
    let targetApi = true;

    // @TODO Change this logic.
    if (adjustedPath.startsWith('/auth')) {
        targetApi = false;
    }

    return `${getApiUrl(targetApi)}${adjustedPath}`;
};

export default class ApiClient {
    makeRequest = (method, path, { query, body } = {}) =>
        new Promise((resolve, reject) => {
            // create request
            const request = superagent[method](formatUrl(path))
            // send cookies from the origin,
            // however only when Access-Control-Allow-Origin is not a wildcard ("*"),
            // and Access-Control-Allow-Credentials is "true".
            .withCredentials();

            if (query) {
                request.query(query);
            }

            // send request
            if (body) {
                request.send(body);
            }

            // resolve or reject promise
            // res.body, res.headers, res.status
            request.end((err, res = {}) => (
                err ? reject(res.body || err) : resolve(res.body)
            ));
        });

    get = this.makeRequest.bind(this, 'get');

    post = this.makeRequest.bind(this, 'post');

    put = this.makeRequest.bind(this, 'put');

    patch = this.makeRequest.bind(this, 'patch');

    del = this.makeRequest.bind(this, 'del');
}
