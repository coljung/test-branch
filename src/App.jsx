import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import i18n from 'i18next';;
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';

// @ssense-ui/components
import { SSENSEThemeProvider } from '@ssense/ui-component-library/lib/styles/SSENSEThemeProvider';
import Header from '@ssense/ui-component-library/lib/layouts/common/Header/Header';
import Sidebar from '@ssense/ui-component-library/lib/layouts/common/Sidebar/Sidebar';
import dashboardStyle from '@ssense/ui-component-library/lib/assets/jss/material-dashboard-pro-react/layouts/dashboardStyle';

// routes
import indexRoutes from './routes/index.js';
import sidebarRoutes from './routes/sidebar.js';

class App extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            mobileOpen: false,
            miniActive: true,
            user: {
                name: 'User Fullname',
            },
            fixedClasses: 'dropdown',
        };
        this.handleDrawerToggle = this.handleDrawerToggle.bind(this);
        this.sidebarMinimize = this.sidebarMinimize.bind(this);
    }

    handleDrawerToggle() {
        this.setState({ mobileOpen: !this.state.mobileOpen });
    }

    sidebarMinimize() {
        this.setState({ miniActive: !this.state.miniActive });
    }

    render() {
        const { classes, ...rest } = this.props;
        const mainPanel =
          `${classes.mainPanel}  ${cx({
            [classes.mainPanelSidebarMini]: this.state.miniActive,
            [classes.mainPanelWithPerfectScrollbar]:
              navigator.platform.indexOf("Win") > -1
          })}`;
        return (
            <SSENSEThemeProvider>
                <BrowserRouter onUpdate={() => window.scrollTo(0, 0)}>
                    <div className={classes.wrapper}>
                        <Sidebar
                        routes={sidebarRoutes}
                        user={this.user}
                        handleDrawerToggle={this.handleDrawerToggle}
                        open={this.state.mobileOpen}
                        miniActive={this.state.miniActive}
                        {...rest} />
                        <div className={mainPanel}>
                            <Header
                            sidebarMinimize={this.sidebarMinimize.bind(this)}
                            miniActive={this.state.miniActive}
                            routes={sidebarRoutes}
                            title={i18n.t('appTitle')}
                            handleDrawerToggle={this.handleDrawerToggle}
                            {...rest} />
                            <main className={classes.content}>
                                <div className={classes.container}>
                                    <Switch>
                                        {indexRoutes.map((route, key) =>
                                            <Route
                                                path={route.path}
                                                render={props => (
                                                    // pass the sub-routes down to keep nesting
                                                    <route.component {...props} />
                                                )}
                                                key={key}
                                                exact={route.exact} />)}
                                    </Switch>
                                </div>
                            </main>
                        </div>
                    </div>
                </BrowserRouter>
            </SSENSEThemeProvider>
        );
    }
}

App.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(dashboardStyle)(App);
