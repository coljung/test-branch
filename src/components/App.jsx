import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Layout, Icon, Spin } from 'antd';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import HeaderContent from './common/HeaderContent';
import CustomNavigation from './customNavigation/CustomNavigation';
import NotificationManager from '../notifications/NotificationManager';
import { me } from '../user/duck/actions';
import { messages } from '../notifications/NotificationActions';
import LoadingSpinner from './common/LoadingSpinner';

class App extends Component {
    static propTypes = {
        location: PropTypes.object,
        children: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.element),
            PropTypes.element,
        ]),
        me: PropTypes.func.isRequired,
        messages: PropTypes.func.isRequired,
        user: PropTypes.object,
    };

    constructor(props) {
        super(props);

        this.state = {
            appReady: false,
            showStoreModal: true,
        };

        // We have to remove the timeout and interval if the class is unmounted
        this.loadTimeout = null;
        this.interval = null;
    }

    componentWillUnmount() {
        clearTimeout(this.loadTimeout);
        clearInterval(this.interval);
        this.setState({ appReady: false });
    }

    componentDidMount() {
        // Wait for ssense client api to be loaded
        this.loadTimeout = setTimeout(() => {
            this.props.messages({ content: 'API NOT LOADED', isError: true });
        }, 5000);
        this.interval = setInterval(() => {
            if (window.ssense) {
                clearTimeout(this.loadTimeout);
                clearInterval(this.interval);

                ssense.onLoad((err) => { // eslint-disable-line no-undef
                    if (err) {
                        this.props.me();
                    } else {
                        this.setState({ appReady: true });
                        this.props.me();
                    }
                });
            }
        }, 250);
    }

    render() {
        const { user } = this.props;

        if (!this.state.appReady) { // And have a user in store
            return (<LoadingSpinner classUsed="loading" />);
        }

        if (!user) { // And have a user in store
            return (<div></div>); // Fast & Compromised: white page that will show the login popup
        }

        const getClassname = this.props.location.pathname === '/' ? 'app_layout_home' : 'app_layout';
        return (
            <div className={getClassname}>
                <Layout>
                    <Layout.Header>
                        <Icon className="trigger" />
                        <HeaderContent />
                    </Layout.Header>
                    <Layout.Sider
                        trigger={null}
                        collapsible
                        collapsed={true}>
                        <CustomNavigation
                            pathname={this.props.location.pathname}
                            />
                    </Layout.Sider>
                    <Layout.Content>
                        <main>
                            {this.props.children}
                            <NotificationManager />
                        </main>
                    </Layout.Content>
                </Layout>
            </div>
        );
    }
}

function mapStateToProps(state) {
    const { userReducer: { user } } = state;

    return {
        user,
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
        me,
        messages,
    }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
