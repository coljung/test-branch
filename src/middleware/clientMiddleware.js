/* global ssense */
import { BACKEND_APP_CODE } from '../constants/common';
// import { messages } from '../notifications/NotificationActions';
import { authenticate as authenticateAction } from '../user/duck/actions';

export default function clientMiddleware(client) {
    return ({ dispatch, getState }) =>
        next => action => { // eslint-disable-line arrow-parens
            if (typeof action === 'function') {
                return action(dispatch, getState);
            }

            const { promise, types, ...rest } = action; // eslint-disable-line no-redeclare
            if (!promise) {
                return next(action);
            }

            const [REQUEST, SUCCESS, FAILURE] = types;
            next({ ...rest, type: REQUEST });

            return promise(client)
                .then(
                    result => next({ ...rest, result, type: SUCCESS }),
                    async (error) => {
                    // if request return 401 show ssense authetication
                        if (error.status === 401) {
                        // await dispatch(authenticateAction()); // eslint-disable-line no-use-before-define
                        // Retry re-dispatch same request and promise again after login
                            dispatch({
                                types: [REQUEST, SUCCESS, FAILURE],
                                promise,
                                ...rest,
                            });
                            return;
                        }

                        // Show error message in UI
                        // dispatch(messages({ isError: true, error }));

                        return next({ ...rest, error, type: FAILURE }); // eslint-disable-line consistent-return
                    },
                )
                .catch(error =>
                    // dispatch(messages({ isError: true, error }));
                    next({ ...rest, error, type: FAILURE }),
                );
        };
}
