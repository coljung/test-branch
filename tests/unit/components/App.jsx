import i18n from 'i18next';
import React from 'react';
import sinon from 'sinon';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ApiClient from '../../../src/ApiClient';
import clientMiddleware from '../../../src/middleware/clientMiddleware';
import App from '../../../src/App';

function setup(state = {}, props = {}) {
    const initialState = {
        userReducer: {
            user: { name: 'Foo Bar' },
        },
        ...state,
    };

    const initialProps = {
        location: {},
        children: [],
        me: jest.fn(),
        user: { name: 'Foo Bars' },
        ...props,
    };

    const client = new ApiClient();
    const middlewares = [thunk, clientMiddleware(client)];

    const store = configureMockStore(middlewares)(initialState);

    return { initialProps, store };
}

let props;

describe('App', () => {
    let mockStore;

    beforeAll(() => {
        props = { location: { pathname: 'pathname' } };
    });

    it('should render correctly', () => {
        const i18nStub = sinon.stub(i18n, 't');
        i18nStub.withArgs('appTitle').returns('UI-Boilerplate');

        const { initialProps, store } = setup();

        const app = renderer.create(
            <App
                location={{ pathname: 'pathname' }}
                store={store}
                {...initialProps} />,
        );
        expect(app).toMatchSnapshot();

        i18nStub.restore();
    });
});
