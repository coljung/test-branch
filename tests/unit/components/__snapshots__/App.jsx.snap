// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="App-wrapper-1"
>
  <div>
    <div
      className="MuiPrivateHiddenCss-smDown-69"
    >
      <div
        className="MuiDrawer-root-79 MuiDrawer-docked-80"
      >
        <div
          className="MuiPaper-root-91 MuiPaper-elevation0-93 MuiDrawer-paper-81 Sidebar-drawerPaper-9 Sidebar-drawerPaperMini-14 MuiDrawer-paperAnchorLeft-82 MuiDrawer-paperAnchorDockedLeft-86"
        >
          <svg
            className="logo"
            viewBox="0 0 540 110"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M482.4,90.6V59.8h48V47.5h-48V19.4h56.2V7h-69.9v96H540V90.6H482.4z M340.4,86.4L295.8,7h-19.2v96h13.7V23.6l44.6,79.4h19.2 V7h-13.7L340.4,86.4z M199.2,59.8h48V47.5h-48V19.4h56.2V7h-69.9v96h71.3V90.6h-57.6V59.8z M411.8,18h1.4c11.7,0,20.4,6.9,21.1,17.8 H448c-0.7-18.5-14.3-30.2-35.5-30.2c-20.7,0-34.3,12.3-34.3,27.4s8.9,23.9,24.3,27l15.4,3.2c10.1,2.1,15.1,6.6,15.1,14.4 s-7,14.4-19.9,14.4h-1.4c-14.4,0-24.5-6.9-24.5-19.9h-13.7c0.7,21.2,15.1,32.2,37.7,32.2c21.2,0,35.5-10.4,35.5-27.3 c0-15.6-9.6-24.4-26.2-27.7l-15.4-3c-8.1-1.6-13.2-6-13.2-13.4C391.9,24.4,399.9,18,411.8,18 M128.5,18h1.4 c11.7,0,20.4,6.9,21.1,17.8h13.7C164,17.3,150.4,5.6,129.1,5.6c-20.7,0-34.3,12.3-34.3,27.4s8.9,23.9,24.3,27l15.4,3.2 c10.1,2.1,15.1,6.6,15.1,14.4s-7,14.4-19.9,14.4h-1.4c-14.4,0-24.5-6.9-24.5-19.9H90.1c0.7,21.2,15.1,32.2,37.7,32.2 c21.2,0,35.5-10.4,35.5-27.3c0-15.6-9.6-24.4-26.2-27.7l-15.4-3c-8.1-1.6-13.2-6-13.2-13.4C108.6,24.4,116.5,18,128.5,18 M38.4,18 h1.4c11.7,0,20.4,6.9,21.1,17.8h13.7C73.9,17.3,60.3,5.6,39.1,5.6C18.4,5.6,4.8,18,4.8,33.1s8.9,23.9,24.3,27l15.4,3.2 c10.1,2.1,15.1,6.6,15.1,14.4S52.5,92,39.6,92h-1.4c-14.4,0-24.5-6.9-24.5-19.9H0c0.7,21.2,15.1,32.2,37.7,32.2 c21.2,0,35.5-10.4,35.5-27.3c0-15.6-9.6-24.4-26.2-27.7l-15.4-3c-8.1-1.6-13.2-6-13.2-13.4C18.5,24.4,26.5,18,38.4,18"
              style={
                Object {
                  "fill": Object {},
                }
              }
            />
          </svg>
          <hr
            className="logo-hr"
          />
          <div
            className="Sidebar-sidebarWrapper-56 Sidebar-drawerPaperMini-14"
          >
            <ul
              className="MuiList-root-118 MuiList-padding-119 Sidebar-list-24"
            >
              <li
                className="MuiListItem-root-122 MuiListItem-default-125 MuiListItem-gutters-130 Sidebar-item-25"
                disabled={false}
              >
                <a
                  aria-current="page"
                  className="Sidebar-itemLink-27  undefined active"
                  href="/"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <div
                    className="MuiListItemIcon-root-134 Sidebar-itemIcon-28 "
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-135"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    className="MuiListItemText-root-144 Sidebar-itemText-30 Sidebar-itemTextMini-33"
                  >
                    Home
                  </div>
                </a>
              </li>
              <li
                className="MuiListItem-root-122 MuiListItem-default-125 MuiListItem-gutters-130 Sidebar-item-25"
                disabled={false}
              >
                <a
                  aria-current={null}
                  className="Sidebar-itemLink-27 "
                  href="/"
                  onClick={[Function]}
                >
                  <div
                    className="MuiListItemIcon-root-134 Sidebar-itemIcon-28 "
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-135"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                      />
                    </svg>
                  </div>
                  <div
                    className="MuiListItemText-root-144 Sidebar-itemText-30 Sidebar-itemTextMini-33"
                  >
                    General
                    <b
                      className="Sidebar-caret-45  "
                    />
                  </div>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="App-mainPanel-2  App-mainPanelSidebarMini-6"
  >
    <header
      className="MuiPaper-root-91 MuiPaper-elevation4-97 MuiAppBar-root-166 MuiAppBar-positionFixed-167 MuiAppBar-colorPrimary-173 mui-fixed Header-appBar-154"
    >
      <div
        className="MuiToolbar-root-175 MuiToolbar-regular-177 MuiToolbar-gutters-176 Header-container-155"
      >
        <div
          className="Header-sidebarMinimize-163"
        >
          <button
            className="MuiButtonBase-root-234 MuiButton-root-208 MuiButton-text-210 MuiButton-flat-213 RegularButton-button-179 RegularButton-transparent-200 RegularButton-justIcon-207"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-209"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-135 Header-sidebarMiniIcon-165"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2 15.5v2h20v-2H2zm0-5v2h20v-2H2zm0-5v2h20v-2H2z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          className="Header-flex-156"
        >
          UI-Boilerplate
        </div>
      </div>
    </header>
    <main
      className="App-content-3"
    >
      <div
        className="App-container-4"
      >
        <div
          className="MuiGrid-container-253"
        >
          <div
            className="MuiGrid-item-254 MuiGrid-grid-xs-12-293 MuiGrid-grid-sm-6-301 MuiGrid-grid-md-6-315 MuiGrid-grid-lg-3-326"
          >
            <div
              className="Card-card-350"
            >
              <div
                className="CardHeader-cardHeader-367 CardHeader-successCardHeader-376 CardHeader-cardHeaderStats-372 CardHeader-cardHeaderIcon-373"
              >
                <div
                  className="CardIcon-cardIcon-381 CardIcon-successCardHeader-383"
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-388"
                  >
                    info_outline
                  </span>
                </div>
                <p
                  className="Home-cardCategory-244"
                >
                  UI-Boilerplate
                </p>
                <h3
                  className="Home-cardTitle-241"
                >
                  Ready
                </h3>
              </div>
              <div
                className="CardFooter-cardFooter-397 CardFooter-cardFooterStats-402"
              >
                <div
                  className="Home-stats-246"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-135"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M21.41 11.58l-9-9C12.05 2.22 11.55 2 11 2H4c-1.1 0-2 .9-2 2v7c0 .55.22 1.05.59 1.42l9 9c.36.36.86.58 1.41.58.55 0 1.05-.22 1.41-.59l7-7c.37-.36.59-.86.59-1.41 0-.55-.23-1.06-.59-1.42zM5.5 7C4.67 7 4 6.33 4 5.5S4.67 4 5.5 4 7 4.67 7 5.5 6.33 7 5.5 7z"
                    />
                  </svg>
                  Let's get cooking
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
